src/core/datastore/deltaset.js

```diff
@@ -0,0 +1,159 @@
+      }
+
+      return deleted;
+    });
+}
+
+function updateDocs(collectionNameWithTag, changed = []) {
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      if (isArray(changed) && changed.length > 0) {
+        return offlineRepo.update(collectionNameWithTag, changed)
+      }
+
+      return changed;
+    });
+}
+
+function makeDeltaSetRequest(collectionNameWithTag, deltaSetQuery, query, options) {
+  const collectionName = stripTagFromCollectionName(collectionNameWithTag);
+  const client = Client.sharedInstance();
+  const request = new KinveyRequest({
+    authType: AuthType.Default,
+    method: RequestMethod.GET,
+    url: format({
+      protocol: client.apiProtocol,
+      host: client.apiHost,
+      pathname: buildCollectionUrl(collectionName, '_deltaset'),
+      query: { since: deltaSetQuery.lastRequest }
+    }),
+    query,
+    timeout: options.timeout,
+    followRedirect: options.followRedirect,
+    cache: options.cache,
+    properties: options.properties,
+    skipBL: options.skipBL,
+    trace: options.trace,
+    client
+  });
+
+  return request.execute()
+    .then((response) => {
+      return updateDeltaSetQuery(deltaSetQuery, response).then(() => response.data);
+    })
+    .then((data) => {
+      return deleteDocs(collectionNameWithTag, data.deleted)
+        .then(() => updateDocs(collectionNameWithTag, data.changed))
+        .then(() => data);
+    });
+}
+
+function makeRequest(collectionNameWithTag, deltaSetQuery, query, options) {
+  if (deltaSetQuery && deltaSetQuery.lastRequest) {
+    if (query && (query.skip != null || query.limit != null)) {
+      throw new KinveyError('You cannot use the skip and limit modifiers on the query when performing a delta set request.');
+    }
+
+    return makeDeltaSetRequest(collectionNameWithTag, deltaSetQuery, query, options);
+  }
+
+  const collectionName = stripTagFromCollectionName(collectionNameWithTag);
+  const client = Client.sharedInstance();
+  const request = new KinveyRequest({
+    authType: AuthType.Default,
+    method: RequestMethod.GET,
+    url: format({
+      protocol: client.apiProtocol,
+      host: client.apiHost,
+      pathname: buildCollectionUrl(collectionName)
+    }),
+    query,
+    timeout: options.timeout,
+    followRedirect: options.followRedirect,
+    cache: options.cache,
+    properties: options.properties,
+    skipBL: options.skipBL,
+    trace: options.trace,
+    client
+  });
+  return request.execute()
+    .then((response) => {
+      return updateDeltaSetQuery(deltaSetQuery, response).then(() => response.data);
+    })
+    .then((data) => {
+      return updateDocs(collectionNameWithTag, data);
+    });
+}
+
+export function deltaSet(collectionNameWithTag, query, options) {
+  return getDeltaSetQuery(collectionNameWithTag, query)
+    .then((deltaSetQuery) => makeRequest(collectionNameWithTag, deltaSetQuery, query, options))
```

Comment:
There was talk about renaming this "query cache" table to "query log" - is this rejected, @tejasranade ? This would make the name of the variable `deltaSetQuery` (and related functions) something like `queryLogEntry`, which is more descriptive in my opinion. Calling it "query" right now is incorrect - it isn't a query, a serialized query is part of it. I know you'll say this is all variable names, but it isn't - it's terms which will be used and they should be well defined.

Reply:
The design refers to the additional table as `_QueryCache`. Since there was some concern about whether it was a "cache", I floated the idea of `_QueryLog` instead. TBH, I suggest we go with `_QueryCache` because that's what we're using in other platforms.

In that context, we could use the names - `getCachedQuery` and `cachedQuery` here.

Reply:
Alright, `_QueryCache` it is. Then all that's left is renaming `deltaSetQuery` to something like "queryCacheEntry", cause `cachedQuery` sounds like what would be in `queryCacheEntry.query`. So just one more term to "coin" - what is the entry in the query cache table :)

Reply:
Okay, `_QueryCache` it is. And I think `getCachedQuery ` and `cachedQuery ` work in this context.

Just one more term to define - what is one entry in that table called? Cause it shouldn't be a "query" (and that's kinda why I didn't want the table to be a cache for queries).